bittensor/dendrite.py

```diff
@@ -120,6 +192,7 @@ def query(
             new_loop = asyncio.new_event_loop()
             asyncio.set_event_loop(new_loop)
             result = loop.run_until_complete(self.forward(*args, **kwargs))
+            self.close_session()
```

Comment:
I am a little confused by this loop, when the try block hits an exception, the default would run the same function again?

Reply:
Also once an error has been hit, this change would now close the session and the connection. Any reason why is necessary? Would it have any effects on non-streaming error handling?

Reply:
The try is for if the current event loop doesn't exist and throws an error, catch is to attempt to create a new loop and run it but the `close_session()` should be in finally. Nice catch!

Reply:
Looks good here.